plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebMarkerResolutionGenerator.java

```diff
@@ -47,6 +45,12 @@
           .equals(marker.getType())) {
         IMarkerResolution fix = new VersionQuickFix();
         markerResolutions[0] = fix;
+      } else if ("com.google.cloud.tools.eclipse.appengine.validation.runtimeMarker"
+          .equals(marker.getType())) {
+        IMarkerResolution fix = new UpgradeRuntimeQuickFix();
+        markerResolutions[0] = fix;
+      } else {
+        return new IMarkerResolution[0];
```

Comment:
I had a comment on this but noticed it got disappeared.

Probably we should do this when an exception occurs, so

```java
      IMarkerResolution fix = new UpgradeRuntimeQuickFix();
      markerResolutions[0] = fix;
    }
    return markerResolutions;
  } catch (CoreException ex) {
    ... // fall through
  }
  return new IMarkerResolution[0]; // <-- should return 0-length array
}
```
and better yet just
```java
     fix = new UpgradeRuntimeQuickFix();
      // markerResolutions[0] = fix;
    }
    return new IMarkerResolution[0] {fix};
  } catch (CoreException ex) {
    ... // fall through
  }
  return new IMarkerResolution[0]; // <-- should return 0-length array when exception
}
```
or even
```java
  try {
    if (...) {
      return new IMarkerResolution[0] {new ApplicationQuickFix()};
    } else if (...) {
      return new IMarkerResolution[0] {new VersionQuickFix()};
    } else if (...) {
      return new IMarkerResolution[0] {new UpgradeRuntimeQuickFix()};
    }
  } catch (CoreException ex) {
    ... // fall through
  }
  return new IMarkerResolution[0]; // <-- should return 0-length array
}
```

Reply:
How about an ArrayList instead?